File: problems/can-scramble/can-scramble.js
const canScramble = (originalString, scrambledString) => {
   if(originalString.length !== scrambledString.length) return false;

   const sortedOriginalStringArray = originalString.split("").sort(),
         sortedScrambledStringArray = scrambledString.split("").sort();

   return sortedOriginalStringArray.every((letter, index) => letter === sortedScrambledStringArray[index]);
}

console.log(canScramble("abc", "cba")); // true
console.log(canScramble("aab", "bba")); // false
console.log(canScramble("xyzxyz", "zzyyxx")); //true
console.log(canScramble("xyzbbxyz", "zzyyxxaa")); //false

File: problems/can-scramble/CanScramble.kt
import kotlin.test.assertFalse
import kotlin.test.assertTrue

fun main() {
    assertTrue("abc".canScramble("cba"))
    assertFalse("ac".canScramble("bb"))
    assertFalse("aab".canScramble("bba"))
}

private fun String.canScramble(other: String) =
    length == other.length && toScrambleMap() == other.toScrambleMap()

private fun String.toScrambleMap() =
    mutableMapOf<Char, Int>().also { forEach { c -> it.merge(c, 1, Int::plus) } }
